src/helpers/map.js

```diff
@@ -174,22 +239,58 @@ export function fillGridWithSubgrid(bigGrid, smallGrid, x, y) {
  */
 export function getTileTraversabilityInCells(tileID) {
   // Start with all cells being traversable
-  const gridTile = init2dArray(CPTL, CPTL, 1);
+  let gridTile = init2dArray(CPTL, CPTL, 1);
 
   if (!isTraversable(tileID)) {
-    const midCell = (CPTL - 1.0) / 2.0;
-    for (let i = 0; i < CPTL; i++) {
-      for (let j = 0; j < CPTL; j++) {
-        const xDiff = Math.max(Math.abs(i - midCell) - 0.5, 0);
-        const yDiff = Math.max(Math.abs(j - midCell) - 0.5, 0);
-        const cellDist = Math.sqrt((xDiff * xDiff) + (yDiff * yDiff));
-        const ppc = PPTL / CPTL; // number of pixels per cell length
-        const pixelDist = cellDist * ppc;
-        if (pixelDist <= getNonTraversableObjectRadius(tileID)) {
-          // This cell touches the object, is not traversable
-          gridTile[i][j] = 0;
+    if (isCNTO(tileID)) {
+      const midCell = (CPTL - 1.0) / 2.0;
+      for (let i = 0; i < CPTL; i++) {
+        for (let j = 0; j < CPTL; j++) {
+          const xDiff = Math.max(Math.abs(i - midCell) - 0.5, 0);
+          const yDiff = Math.max(Math.abs(j - midCell) - 0.5, 0);
+          const cellDist = Math.sqrt((xDiff * xDiff) + (yDiff * yDiff));
+          const ppc = PPTL / CPTL; // number of pixels per cell length
+          const pixelDist = cellDist * ppc;
+          if (pixelDist <= getCNTORadius(tileID)) {
+            // This cell touches the object, is not traversable
+            gridTile[i][j] = 0;
+          }
+        }
+      } // tile is noncircular and nontraversable
+    } else if (tileID === tileTypes.ANGLE_WALL_1) {
```

Comment:
I haven't looked at this too closely, but is there a way to consolidate this code? You repeat a lot of the same logic